packages/model/src/reminder.ts

```diff
@@ -0,0 +1,48 @@
+import { ListResponse } from './common';
+import { ResearchOutputResponse } from './research-output';
+
+type ReminderArea = 'Research Output' | 'Events';
+
+export const reminderTypes = [
+  'ResearchOutputPublished',
+  'ResearchOutputShare',
+] as const;
+
+export type ReminderType = typeof reminderTypes[number];
+
+interface Reminder {
+  area: ReminderArea;
+  type: ReminderType;
+  data: unknown;
+}
+
+export interface ResearchOutputPublishedReminder extends Reminder {
+  area: 'Research Output';
+  type: 'ResearchOutputPublished';
+  data: {
+    researchOutput: ResearchOutputResponse;
+  };
+}
+
+export interface ResearchOutputShareReminder extends Reminder {
+  area: 'Research Output';
+  type: 'ResearchOutputShare';
+}
+
+export type ReminderResponse =
+  | ResearchOutputPublishedReminder
+  | ResearchOutputShareReminder;
+
+export type ListReminderResponse = ListResponse<ReminderResponse>;
+
+export const getReminderCard = (reminder: ReminderResponse) => {
```

Comment:
Not sure about including code in the model? Shouldn't we just have a type discriminator and handle this somewhere else?

Reply:
this is just an example code to show how to recognise which reminder we are dealing with. This way once you recognise the type you can access the data easily